lib/SIL/OwnershipUtils.cpp

```diff
@@ -125,3 +173,34 @@ bool swift::getUnderlyingBorrowIntroducers(SILValue inputValue,
 
   return true;
 }
+
+llvm::raw_ostream &swift::operator<<(llvm::raw_ostream &os,
+                                     BorrowScopeIntroducerKind kind) {
+  kind.print(os);
+  return os;
+}
+
+bool BorrowScopeIntroducer::areInstructionsWithinScope(
```

Comment:
yes. I am with you. I just wanted to abstract this a way since it is a general operation we have been performing. I can abstract it out.

Reply:
@atrick for some cases I don't think it will go away, but in a subsequent commit, I have a small helper that in the case where we /know/ that we are never going to pass a cond_br we can just convert the ArrayRef<SILInstruction *> to ArrayRef<BranchPropagatedUser>. The reason why this is safe is BranchPropagatedUser is just a pointer int pair internally and if it is not storing a cond_br, then it is exactly bitwise the same as the underlying SILInstruction *. I can make the helper safe to use by performing a dynamic check if we have SILInstruction and a compile time check if we have an ArrayRef<SILInstructionSubclass>.

Reply:
That fact that `cond_br` still has arguments is dumb but it's unrelated to this PR. I'm just anxious for that to go away so code like this can be vastly simpler.